packages/theme-language-server-common/src/documentLinks/DocumentLinksProvider.ts

```diff
@@ -91,3 +129,149 @@ function range(textDocument: TextDocument, node: { position: LiquidHtmlNode['pos
 function isLiquidString(node: LiquidHtmlNode): node is LiquidString {
   return node.type === NodeTypes.String;
 }
+
+function createDocumentLinkForTypeNode(
+  typeNode: JSONNode,
+  parentNode: LiquidRawTag,
+  textDocument: TextDocument,
+  root: URI,
+  blockType: string,
+): DocumentLink | null {
+  const startOffset = typeNode.offset;
+  const endOffset = typeNode.offset + typeNode.length;
+  const startPos = parentNode.body.position.start + startOffset;
+  const endPos = parentNode.body.position.start + endOffset;
+
+  const start = textDocument.positionAt(startPos);
+  const end = textDocument.positionAt(endPos);
+
+  return DocumentLink.create(
+    Range.create(start, end),
+    Utils.resolvePath(root, 'blocks', `${blockType}.liquid`).toString(),
+  );
+}
+
+function processPresetBlocks(
```

Comment:
This is kind of a hefty function. it would help with scanability to encapsulate common functionality between the conditional branches.